Buildings/Controls/OBC/CDL/Logical/Sources/TimeTable.mo

```diff
@@ -0,0 +1,235 @@
+within Buildings.Controls.OBC.CDL.Logical.Sources;
+block TimeTable
+  "Table look-up with respect to time with constant segments"
+
+  parameter Real table[:,:]
+  "Table matrix with time as a first column (in seconds, unless timeScale is not 1) and Booleans in all other columns";
+  parameter CDL.Types.Extrapolation extrapolation=CDL.Types.Extrapolation.HoldLastPoint
+    "Extrapolation of data outside the definition range";
+  parameter Modelica.SIunits.Time timeScale=1
+    "Time scale of first table column. Set to 3600 if time in table is in hours";
+
+  Interfaces.BooleanOutput y[nout] "Output of the table"
+    annotation (Placement(transformation(extent={{120,-20},{160,20}}),
+        iconTransformation(extent={{100,-20},{140,20}})));
+
+protected
+  final parameter Integer nout=size(table, 2)-1
+    "Dimension of output vector";
+  final parameter Integer n=size(table, 1)
+    "Number of table points";
+  parameter Modelica.SIunits.Time t0(fixed=false)
+    "First sample time instant";
+
+  Integers.Equal intEquFal[nout] "Check values equal false"
+    annotation (Placement(transformation(extent={{-40,70},{-20,90}})));
+  Integers.Equal intEquTru[nout] "Checks values equal true"
+    annotation (Placement(transformation(extent={{-40,40},{-20,60}})));
+  Integers.Sources.Constant intFal[nout](k=fill(0, nout)) "Integer false"
+    annotation (Placement(transformation(extent={{-80,70},{-60,90}})));
+  Integers.Sources.Constant intTru[nout](k=fill(1, nout)) "Integer true"
+    annotation (Placement(transformation(extent={{-80,40},{-60,60}})));
+  Or or2[nout] "Checks if the table value is either true or false"
+    annotation (Placement(transformation(extent={{20,60},{40,80}})));
+  MultiAnd mulAnd(
+    final nu=nout) "Multi and"
+    annotation (Placement(transformation(extent={{50,60},{70,80}})));
+  MultiOr mulOr1(
+    final nu=nout) "Multi or"
+    annotation (Placement(transformation(extent={{20,-100},{40,-80}})));
+  And and2
+    "Logical and"
+    annotation (Placement(transformation(extent={{60,20},{80,40}})));
+  Buildings.Controls.OBC.CDL.Conversions.RealToInteger realToInteger[nout]
+    "Converts scheduled values to integer"
```

Comment:
Move the test for the data type to `initial equation` because this test only depends on parameter values.